File: Data Structures & Algorithms/is-subsequence/submission-0.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() > t.size()) {
            return false;
        }
        
        int l = 0;
        for (int i = 0; i < t.size(); i++) {
            if (s[l] == t[i]) {
                l++;
            }
            if (s.size() == l) {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        hashmap = {")" : "(", "}" : "{", "]" : "["}
        stack = []


        for i in s:
            if i not in hashmap:
                stack.append(i)
                continue
                # stack.append(i)
            if not stack or stack[-1] != hashmap[i]:
                return False

            stack.pop()

        return not stack